Unable to terraform import `aiven_aws_vpc_peering_connection` resources

When I try to run

```
terraform import aiven_aws_vpc_peering_connection.my_connections[\"vpc-012345\"] my_project/my_vpc/my_aws_account_id/vpc-012345
```
I see the following error:

```
Error: invalid identifier my_project/my_vpc/my_aws_account_id/vpc-012345, expected <project_name>/<vpc_id>/<aws_account_id>
```

Looking at the documentation this matches the suggested [import code](https://registry.terraform.io/providers/aiven/aiven/latest/docs/resources/aws_vpc_peering_connection#import). However, trying
```
terraform import aiven_aws_vpc_peering_connection.my_connections my_project/my_vpc/my_aws_account_id
```
gives the following error:

```
aiven_aws_vpc_peering_connection.my_connections: Importing from ID "my_project/my_vpc/my_aws_account_id"...
|
| Error: Plugin did not respond
| 
| The plugin encountered an error, and failed to respond to the plugin.(*GRPCProvider).ImportResourceState call. The plugin logs may contain more details.
|


Stack trace from the terraform-provider-aiven_v3.0.0 plugin:

panic: runtime error: index out of range [3] with length 3

goroutine 71 [running]:
github.com/aiven/terraform-provider-aiven/internal/service/vpc.parsePeeringVPCId(0xc0000484c0, 0x39, 0xc0005114d8, 0x44e30c, 0x30, 0xcde660, 0x1, 0xc0005e4930, 0xc000511530, 0x4f79ff, ...)
        github.com/aiven/terraform-provider-aiven/internal/service/vpc/resource_vpc_peering_connection.go:234 +0x1d6
github.com/aiven/terraform-provider-aiven/internal/service/vpc.resourceAWSVPCPeeringConnectionRead(0xf769e0, 0xc0005e4780, 0xc000a25400, 0xdc2ba0, 0xc0001906e0, 0x2, 0xc0005e4930, 0x3)
        github.com/aiven/terraform-provider-aiven/internal/service/vpc/resource_aws_vpc_peering_connection.go:160 +0x97
github.com/aiven/terraform-provider-aiven/internal/service/vpc.resourceAWSVPCPeeringConnectionImport(0xf769e0, 0xc0005e4780, 0xc000a25400, 0xdc2ba0, 0xc0001906e0, 0x12, 0x0, 0x0, 0x0, 0x20)
        github.com/aiven/terraform-provider-aiven/internal/service/vpc/resource_aws_vpc_peering_connection.go:228 +0x259
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Provider).ImportState(0xc000114360, 0xf769e0, 0xc0005e4780, 0xc000511768, 0xc0000484c0, 0x39, 0x9, 0xc000000001, 0xe216ec, 0x12, ...)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/provider.go:385 +0x544
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ImportResourceState(0xc00098c360, 0xf76938, 0xc0005e4780, 0xc000c24fe0, 0xe216ec, 0x12, 0x0)
        github.com/hashicorp/terraform-plugin-sdk/v2@v2.16.0/helper/schema/grpc_provider.go:1088 +0x11a
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ImportResourceState(0xc000176000, 0xf769e0, 0xc0005e4060, 0xc0005e22d0, 0x0, 0x0, 0x0)
        github.com/hashicorp/terraform-plugin-go@v0.9.0/tfprotov5/tf5server/server.go:841 +0x2ed
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ImportResourceState_Handler(0xddfb20, 0xc000176000, 0xf769e0, 0xc0005e4060, 0xc000115200, 0x0, 0xf769e0, 0xc0005e4060, 0xc00073e900, 0x5d)
        github.com/hashicorp/terraform-plugin-go@v0.9.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:403 +0x214
google.golang.org/grpc.(*Server).processUnaryRPC(0xc00013e540, 0xf7e598, 0xc0003c81a0, 0xc0009d9680, 0xc0003cb1d0, 0x1435358, 0x0, 0x0, 0x0)
        google.golang.org/grpc@v1.46.0/server.go:1283 +0x544
google.golang.org/grpc.(*Server).handleStream(0xc00013e540, 0xf7e598, 0xc0003c81a0, 0xc0009d9680, 0x0)
        google.golang.org/grpc@v1.46.0/server.go:1620 +0xd0c
google.golang.org/grpc.(*Server).serveStreams.func1.2(0xc0002ba000, 0xc00013e540, 0xf7e598, 0xc0003c81a0, 0xc0009d9680)
        google.golang.org/grpc@v1.46.0/server.go:922 +0xab
created by google.golang.org/grpc.(*Server).serveStreams.func1
        google.golang.org/grpc@v1.46.0/server.go:920 +0x1fd

Error: The terraform-provider-aiven_v3.0.0 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

I've tried a number of different combinations of the command, but all result in an error similar to this.

The import command that I used for the older version of this resource `aiven_vpc_peering_connection` used the syntax that I tried first, which is why I tried that first. Looking at the import command for similar commands for [Azure](https://registry.terraform.io/providers/aiven/aiven/latest/docs/resources/azure_vpc_peering_connection#import) and [GCP](https://registry.terraform.io/providers/aiven/aiven/latest/docs/resources/gcp_vpc_peering_connection#import) also seem to show that the peering connection ID is required.

It looks like this is missing from the `aiven_aws_vpc_peering_connection` resource.

--- a/docs/resources/aws_vpc_peering_connection.md
+++ b/docs/resources/aws_vpc_peering_connection.md
@@ -27,11 +27,11 @@ resource "aiven_aws_vpc_peering_connection" "foo" {
 
 - **aws_account_id** (String) AWS account ID. This property cannot be changed, doing so forces recreation of the resource.
 - **aws_vpc_id** (String) AWS VPC ID. This property cannot be changed, doing so forces recreation of the resource.
+- **aws_vpc_region** (String) AWS region of the peered VPC (if not in the same region as Aiven VPC). This property cannot be changed, doing so forces recreation of the resource.
 - **vpc_id** (String) The VPC the peering connection belongs to. This property cannot be changed, doing so forces recreation of the resource.
 
 ### Optional
 
-- **aws_vpc_region** (String) AWS region of the peered VPC (if not in the same region as Aiven VPC). This property cannot be changed, doing so forces recreation of the resource.
 - **id** (String) The ID of this resource.
 - **timeouts** (Block, Optional) (see [below for nested schema](#nestedblock--timeouts))
 
@@ -54,5 +54,5 @@ Optional:
 Import is supported using the following syntax:
 
 ```shell
-terraform import aiven_aws_vpc_peering_connection.foo project_name/vpc_id/aws_account_id
+terraform import aiven_aws_vpc_peering_connection.foo project_name/vpc_id/aws_account_id/aws_vpc_id/aws_vpc_region
 ```

--- a/examples/resources/aiven_aws_vpc_peering_connection/import.sh
+++ b/examples/resources/aiven_aws_vpc_peering_connection/import.sh
@@ -1,1 +1,1 @@
-terraform import aiven_aws_vpc_peering_connection.foo project_name/vpc_id/aws_account_id
+terraform import aiven_aws_vpc_peering_connection.foo project_name/vpc_id/aws_account_id/aws_vpc_id/aws_vpc_region

--- a/internal/service/vpc/resource_aws_vpc_peering_connection.go
+++ b/internal/service/vpc/resource_aws_vpc_peering_connection.go
@@ -37,7 +37,7 @@ var aivenAWSVPCPeeringConnectionSchema = map[string]*schema.Schema{
 	},
 	"aws_vpc_region": {
 		ForceNew: true,
-		Optional: true,
+		Required: true,
 		Type:     schema.TypeString,
 		DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
 			return new == ""
@@ -215,19 +215,27 @@ func resourceAWSVPCPeeringConnectionDelete(ctx context.Context, d *schema.Resour
 		MinTimeout: 2 * time.Second,
 	}
 	if _, err := stateChangeConf.WaitForStateContext(ctx); err != nil && !aiven.IsNotFound(err) {
-		return diag.Errorf("Error waiting for Aiven AWS VPC Peering Connection to be DELETED: %s", err)
+		return diag.Errorf("Error waiting for AWS Aiven VPC Peering Connection to be DELETED: %s", err)
 	}
 	return nil
 }
 
 func resourceAWSVPCPeeringConnectionImport(ctx context.Context, d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
-	if len(strings.Split(d.Id(), "/")) != 3 {
-		return nil, fmt.Errorf("invalid identifier %v, expected <project_name>/<vpc_id>/<aws_account_id>", d.Id())
+	if len(strings.Split(d.Id(), "/")) != 5 {
+		return nil, fmt.Errorf("invalid identifier %v, expected <project_name>/<vpc_id>/<aws_account_id>/<aws_vpc_id>/<aws_vpc_region>", d.Id())
+	}
+
+	client := m.(*aiven.Client)
+
+	projectName, vpcID, peerCloudAccount, peerVPC, peerRegion := parsePeeringVPCId(d.Id())
+	_, err := client.VPCPeeringConnections.GetVPCPeering(projectName, vpcID, peerCloudAccount, peerVPC, peerRegion)
+	if err != nil && schemautil.IsUnknownResource(err) {
+		return nil, errors.New("cannot find specified VPC peering connection")
 	}
 
 	dig := resourceAWSVPCPeeringConnectionRead(ctx, d, m)
 	if dig.HasError() {
-		return nil, errors.New("cannot get aws vpc peering connection")
+		return nil, errors.New("cannot get AWS VPC peering connection")
 	}
 
 	return []*schema.ResourceData{d}, nil

--- a/internal/service/vpc/resource_azure_vpc_peering_connection.go
+++ b/internal/service/vpc/resource_azure_vpc_peering_connection.go
@@ -235,7 +235,7 @@ func resourceAzureVPCPeeringConnectionDelete(ctx context.Context, d *schema.Reso
 		MinTimeout: 2 * time.Second,
 	}
 	if _, err := stateChangeConf.WaitForStateContext(ctx); err != nil && !aiven.IsNotFound(err) {
-		return diag.Errorf("Error waiting for Aiven VPC Peering Connection to be DELETED: %s", err)
+		return diag.Errorf("Error waiting for Azure Aiven VPC Peering Connection to be DELETED: %s", err)
 	}
 	return nil
 }
@@ -243,6 +243,14 @@ func resourceAzureVPCPeeringConnectionDelete(ctx context.Context, d *schema.Reso
 func resourceAzureVPCPeeringConnectionImport(ctx context.Context, d *schema.ResourceData, m interface{}) ([]*schema.ResourceData, error) {
 	if len(strings.Split(d.Id(), "/")) != 4 {
 		return nil, fmt.Errorf("invalid identifier %v, expected <project_name>/<vpc_id>/<azure_subscription_id>/<vnet_name>", d.Id())
+	}
+
+	client := m.(*aiven.Client)
+
+	projectName, vpcID, peerCloudAccount, peerVPC, peerRegion := parsePeeringVPCId(d.Id())
+	_, err := client.VPCPeeringConnections.GetVPCPeering(projectName, vpcID, peerCloudAccount, peerVPC, peerRegion)
+	if err != nil && schemautil.IsUnknownResource(err) {
+		return nil, errors.New("cannot find specified Azure VPC peering connection")
 	}
 
 	dig := resourceAzureVPCPeeringConnectionRead(ctx, d, m)

--- a/internal/service/vpc/resource_gcp_vpc_peering_connection.go
+++ b/internal/service/vpc/resource_gcp_vpc_peering_connection.go
@@ -206,9 +206,17 @@ func resourceGCPVPCPeeringConnectionImport(ctx context.Context, d *schema.Resour
 		return nil, fmt.Errorf("invalid identifier %v, expected <project_name>/<vpc_id>/<gcp_project_id>/<peer_vpc>", d.Id())
 	}
 
+	client := m.(*aiven.Client)
+
+	projectName, vpcID, peerCloudAccount, peerVPC, peerRegion := parsePeeringVPCId(d.Id())
+	_, err := client.VPCPeeringConnections.GetVPCPeering(projectName, vpcID, peerCloudAccount, peerVPC, peerRegion)
+	if err != nil && schemautil.IsUnknownResource(err) {
+		return nil, errors.New("cannot find specified GCP VPC peering connection")
+	}
+
 	dig := resourceGCPVPCPeeringConnectionRead(ctx, d, m)
 	if dig.HasError() {
-		return nil, errors.New("cannot get VPC peering connection")
+		return nil, errors.New("cannot get GCP VPC peering connection")
 	}
 
 	return []*schema.ResourceData{d}, nil

--- a/internal/service/vpc/resource_vpc_peering_connection.go
+++ b/internal/service/vpc/resource_vpc_peering_connection.go
@@ -365,6 +365,14 @@ func resourceVPCPeeringConnectionImport(ctx context.Context, d *schema.ResourceD
 		return nil, fmt.Errorf("invalid identifier %v, expected <project_name>/<vpc_id>/<peer_cloud_account>/<peer_vpc>", d.Id())
 	}
 
+	client := m.(*aiven.Client)
+
+	projectName, vpcID, peerCloudAccount, peerVPC, peerRegion := parsePeeringVPCId(d.Id())
+	_, err := client.VPCPeeringConnections.GetVPCPeering(projectName, vpcID, peerCloudAccount, peerVPC, peerRegion)
+	if err != nil && schemautil.IsUnknownResource(err) {
+		return nil, errors.New("cannot find specified VPC peering connection")
+	}
+
 	dig := resourceVPCPeeringConnectionRead(ctx, d, m)
 	if dig.HasError() {
 		return nil, errors.New("cannot get VPC peering connection")

